presto-cli/src/main/java/com/facebook/presto/cli/AbstractWarningsPrinter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cli;
+
+import com.facebook.presto.spi.PrestoWarning;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.cli.ConsolePrinter.REAL_TERMINAL;
+import static java.lang.String.format;
+
+abstract class AbstractWarningsPrinter
+        implements WarningsPrinter
+{
+    private static final String WARNING_BEGIN = ((char) 27) + "[33m";
+    private static final String WARNING_END = ((char) 27) + "[39m";
+
+    private boolean hasProcessedWarnings;
+    private int processedWarningsCount;
+
+    protected String getWarningMessage(PrestoWarning warning)
+    {
+        // If this is a real terminal color the warnings yellow
+        return REAL_TERMINAL ? format("%sWARNING: %s%s", WARNING_BEGIN, warning.getMessage(), WARNING_END) :
+                format("WARNING: %s", warning.getMessage());
+    }
+
+    protected int getProcessedWarningsCount()
+    {
+        return processedWarningsCount;
+    }
+
+    protected Optional<PrestoWarning> nextWarning(List<PrestoWarning> warnings)
```

Comment:
you can consider returning an `Iterator<PrestoWarning>` here (you can use `com.google.common.collect.AbstractIterator`).